File: selfstudy/python/books_reverse.py
'''
task 158:
Books - Reverse
You need to reverse the order of book page counts
'''

def reverse_order_books(pages):
    return pages[::-1]

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], reverse_order_books([10, 20, 30])),
        ([50, 60, 70], reverse_order_books([50, 60, 70])),
        ([1, 2, 3], reverse_order_books([1, 2, 3])),
        ([100, 200], reverse_order_books([100, 200])),
        ([], reverse_order_books([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = reverse_order_books(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# done

File: selfstudy/golang/books_reverse.go
/*
task 258:
Books - Reverse
You need to reverse the order of book page counts
*/

package main

import (
	"fmt"
	"reflect"
)

func Reverse_orderBooks(pages []int) []int {
	res := make([]int, len(pages))
	for i, v := range pages {
		res[len(pages)-1-i] = v
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Reverse_orderBooks([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Reverse_orderBooks([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Reverse_orderBooks([]int{1, 2, 3})},
		{[]int{100, 200}, Reverse_orderBooks([]int{100, 200})},
		{[]int{}, Reverse_orderBooks([]int{})},
	}
	
	for idx, t := range tests {
		res := Reverse_orderBooks(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// works i think
